main.go

```diff
@@ -135,3 +319,183 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", TBD_VERSION_HEADER_REGEX, err)
+	}
+	breakingChangesRegex, err := regexp.Compile(BREAKING_CHANGES_SUBHEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", BREAKING_CHANGES_SUBHEADER_REGEX, err)
+	}
+	versionHeaderRegex, err := regexp.Compile(VERSION_HEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", VERSION_HEADER_REGEX, err)
+	}
+
+    scanner := bufio.NewScanner(changelogFile)
+	// Find TBD header
+    for scanner.Scan() {
+		if tbdRegex.Match(scanner.Bytes()) {
+			break
+		}
+    }
+	// Scan file until next version header detected, searching for Breaking Changes header along the way
+	foundBreakingChanges := false
+	for scanner.Scan() {
+		if breakingChangesRegex.Match(scanner.Bytes()){
+			foundBreakingChanges = true
+		}
+		
+		if versionHeaderRegex.Match(scanner.Bytes()){
+			break
+		}
+	}
+
+    if err := scanner.Err(); err != nil {
+        logrus.Errorf("An error occurred while scanning file.\n", err)
+    }
+    return foundBreakingChanges
+}
+
+func runPreReleaseScripts(preReleaseScriptsDirpath string, releaseVersion string) error {
+	preReleaseScriptsFilepath := path.Join(preReleaseScriptsDirpath, preReleaseScriptsFilename)
+	preReleaseScriptsFile, err := os.Open(preReleaseScriptsFilepath);
+	if err != nil {
+		logrus.Errorf("An error occurred attempting to open file at provided path. Are you sure '%s' exists?", preReleaseScriptsFilepath, err)
+		return err
+	}
+	defer preReleaseScriptsFile.Close()
+	scanner := bufio.NewScanner(preReleaseScriptsFile)
+	var scripts []string
+	for scanner.Scan() {
+		scripts = append(scripts, scanner.Text())
+	}
+	for _, script := range scripts {
+		scriptCmdString := path.Join(preReleaseScriptsDirpath, script)
+		scriptCmd := exec.Command(scriptCmdString, releaseVersion)
+		err := scriptCmd.Run()
+		if err != nil {
+			logrus.Errorf("An error occurred attempting to run the following pre release script command: '%s'", scriptCmdString, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func updateChangelog(changelogFilepath string, releaseVersion string) error {
+	clFile, err := os.ReadFile(changelogFilepath)
+	if err != nil {
+		logrus.Errorf("Error attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath, err)
+	}
+	tbdRegex, err := regexp.Compile(TBD_VERSION_HEADER_REGEX)
+	if err != nil {
+		logrus.Errorf("Could not parse regexp: '%s'", TBD_VERSION_HEADER_REGEX, err)
+	}
+	lines := bytes.Split(clFile, []byte("\n"))
+	newLines:= make([][]byte, len(lines) + 1) 
+	// Add a new TBD head for next release
+	newLines[0] = []byte("# TBD")
+	for i, line := range lines {
+		// Change current TBD header to Release Version header
+		if tbdRegex.Match(line){
+			releaseVersionHeader := fmt.Sprintf("# %s", releaseVersion)
+			newLines[i + 1] = []byte(releaseVersionHeader)
+		}
+		if i == 0|| i == len(newLines) - 1{
```

Comment:
Whoa won't this mean we're deleting the 0th and the last line?

Reply:
Got rid of this checks by using a more intuitive counter.